OpenRA.Game/Graphics/ModelRenderer.cs

```diff
@@ -163,64 +164,70 @@ public void SetViewportParams(Size screen, float zoom, int2 scroll)
 			CalculateSpriteGeometry(tl, br, 1, out spriteSize, out spriteOffset);
 			CalculateSpriteGeometry(stl, sbr, 2, out shadowSpriteSize, out shadowSpriteOffset);
 
-			if (sheetBuilderForFrame == null)
-				sheetBuilderForFrame = new SheetBuilder(SheetType.BGRA, AllocateSheet);
-
-			var sprite = sheetBuilderForFrame.Allocate(spriteSize, 0, spriteOffset);
-			var shadowSprite = sheetBuilderForFrame.Allocate(shadowSpriteSize, 0, shadowSpriteOffset);
-			var sb = sprite.Bounds;
-			var ssb = shadowSprite.Bounds;
-			var spriteCenter = new float2(sb.Left + sb.Width / 2, sb.Top + sb.Height / 2);
-			var shadowCenter = new float2(ssb.Left + ssb.Width / 2, ssb.Top + ssb.Height / 2);
-
-			var translateMtx = Util.TranslationMatrix(spriteCenter.X - spriteOffset.X, renderer.SheetSize - (spriteCenter.Y - spriteOffset.Y), 0);
-			var shadowTranslateMtx = Util.TranslationMatrix(shadowCenter.X - shadowSpriteOffset.X, renderer.SheetSize - (shadowCenter.Y - shadowSpriteOffset.Y), 0);
-			var correctionTransform = Util.MatrixMultiply(translateMtx, FlipMtx);
-			var shadowCorrectionTransform = Util.MatrixMultiply(shadowTranslateMtx, ShadowScaleFlipMtx);
-
-			doRender.Add(Pair.New<Sheet, Action>(sprite.Sheet, () =>
+			Sprite sprite = null;
+			Sprite shadowSprite = null;
```

Comment:
If not in a frame, I've chosen to leave the sprites null here. We could still set them if required, it's just the values inside them referencing where they live on the sheet would be bogus since the sheet will be reused when the next frame starts.